sdk/data/azappconfig/snapshot.go

```diff
@@ -0,0 +1,52 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azappconfig
+
+import (
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type Snapshot struct {
+	// REQUIRED; A list of filters used to filter the key-values included in the snapshot.
+	Filters []KeyValueFilter
```

Comment:
Is this useful info to return? IIRC, part of the problem with the generated `Snapshot` type is that it combines input and output state. Since our `Snapshot` type is output-only, we should prune all fields that are specific to input.

Reply:
Yes we only require the name and status fields

Reply:
Discussed offline. Reverted changes. The struct requires these fields for response. An additional struct for SnapshotInfo has been added to handle response from BeginCreateSnapshot.